docs/assessment.md

```diff
@@ -126,8 +130,113 @@ By default, PrairieLearn provides an incentive for students to answer a specific
+2. If the `submission percentage score` is 100%, then reset `best current score` to zero. Otherwise, if the `submission percentage score` is more than `best current score`, then set `best current score` to `submission percentage score`.
+3. If the `submission percentage score` is 100% and `constantQuestionValue` is not set to true, then increase `question value` by `autoPoints`. Otherwise set `question value` to `autoPoints`.
+
+As an example, suppose a student makes 8 submissions of varying degrees of correctness to a question. Assume the question has `autoPoints` set to 4 and `maxAutoPoints` set to 10. The student's submissions are as follows:
+
+| Submission number | Submission perc. score | Best current score | Question value | Question points | Comment                                                                                                   |
+| ----------------- | ---------------------- | ------------------ | -------------- | --------------- | --------------------------------------------------------------------------------------------------------- |
+| Init              | -                      | 0                  | 4              | 0               | Initial values                                                                                            |
+| 1                 | 50%                    | 50%                | 4              | 2               | (50% - 0%) \* 4 = 2 points awarded                                                                        |
+| 2                 | 80%                    | 80%                | 4              | 3.2             | (80% - 50%) \* 4 = 1.2 points awarded                                                                     |
+| 3                 | 20%                    | 80%                | 4              | 3.2             | No improvement in score, so no points awarded or deducted                                                 |
+| 4                 | 100%                   | 0%                 | 8              | 4               | (100% - 80%) \* 4 = 0.8 points awarded, `best current score` reset to 0%, `question value` increased to 8 |
+| 5                 | 50%                    | 50%                | 4              | 8               | (50% - 0%) \* 8 = 4 points awarded, `question value` reset to 4                                           |
+| 6                 | 0%                     | 50%                | 4              | 8               | No improvement in score, so no points awarded or deducted                                                 |
+| 7                 | 90%                    | 90%                | 4              | 9.6             | (90% - 50%) \* 4 = 1.6 points awarded or deducted                                                         |
+| 8                 | 100%                   | 0%                 | 8              | 10              | (100% - 90%) \* 4 = 0.4 points awarded, `best current score` reset to 0%, `question value` increased to 8 |
```

Comment:
I'd suggest adding a new submission with 100%, so that the increased question value can be seen in practice. This would imply increasing the max points to 18, and updating the following paragraph. Alternatively, max points could be set to 16, so that the cap can be used as well.

Reply:
Added in 8a13d590e9b8533f33ce23a2df9beb1e877c13da